### PR TITLE
Contracts: add clean post-install script

### PR DESCRIPTION
**Pull request by justinkaseman**
### Description

After adding `hardhat-upgrades` as a dev dependency to use within contract unit tests, there have been issues during dev work when running `pnpm compile` caused by `proper-lockfile`: "Error: Lock file is already being held"

This PR adds a `postinstall` script, which is ran before installs to clear the contract lockfile.

This must be ignored when running `pnpm i --prod`, because there is no `hardhat` package installed.

### CODE REVIEW

**Review by jmank88 — DISMISSED** [review]

**Review by samsondav — DISMISSED** [review]

**Comment by justinkaseman** [comment]
Waiting on Release Engineering team's feedback before merging

File: .github/actions/setup-nodejs/action.yaml
Comment (HenryNguyen5) [inline .github/actions/setup-nodejs/action.yaml:29]:
```suggestion
      run: pnpm i --prod --ignore-scripts
```

Don't think we need the boolean here

**Review by HenryNguyen5 — CHANGES_REQUESTED** [review]
I don't think this is how we want to accomplish clean installs. Per the `pnpm` documentation:
```
Do not execute any scripts defined in the project package.json and its **dependencies**.
```

This is going to possibly change project behaviour during prod since we'll be skipping dep lifecycle scripts too, not just our own.

I've opened up https://github.com/OpenZeppelin/openzeppelin-upgrades/pull/769 to allow for configurable retries, which should solve the crux of the issue which is high lockfile contention causing timeouts

**Comment by justinkaseman** [comment]
Closing in favor of hardhat-upgrades PR

**Pull request closed without merging**
